### DISCUSSION TITLE
Make `editors` its own separate package?

### DISCUSSION THREAD

**Discussion by NucciTheBoss** [j4m-can/hpct-library#11] [Ideas]
So I have been thinking about how we can further optimize `hpctlib` since "bloat" seems to be one of the bigger concerns about the library. This sprint I am working with the conf file editor for slurm, and I had a thought. What if we broke the editors off into their own package? Something like `confeditors`, `hpcteditors`, etc.

Reason saying is that looking through the code base, the editors seem to be relatively isolated from the rest of `hpctlib`. Also, by breaking the editors off into their own package, we can then pull in the editors into programs that need to edit the conf files, but don't necessarily need the `ops`, `misc`, `interface`, or `os` modules. For example, we could write a client-side interview program that bootstraps a slurm.conf file and then passes it off to `Configurator` for bringing up new nodes.

Since this is an idea, I am looking for the following feed back:

* Pros - why breaking editors off into their own package could be good for `hpctlib`?
* Cons - why might we want to keep editors in `hpctlib`?

**Comment by j4m-can** [comment]
`hpctlib` started off with mostly charms in mind. I.e., that they would be used in charms. But now that we are dealing with managers which themselves can be developed and run apart from charms, I think it is time to split them out. So, to me this is the winning proposition here.

I'll plan for this for the next sprint, unless I somehow how a lot of free time.

**Reply by NucciTheBoss** [reply]
Good to see that we are on the same page. I could potentially look next week provided that the slurm charms behave as expected.

**Comment by j4m-can** [comment]
https://github.com/j4m-can/hpct-editors now exists.
